core/src/main/java/org/apache/iceberg/view/BaseMetastoreViewCatalog.java

```diff
@@ -0,0 +1,190 @@
+    super.initialize(name, properties);
+  }
+
+  @Override
+  public String name() {
+    return super.name();
+  }
+
+  @Override
+  public View loadView(TableIdentifier identifier) {
+    if (isValidIdentifier(identifier)) {
+      ViewOperations ops = newViewOps(identifier);
+      if (ops.current() == null) {
+        throw new NoSuchViewException("View does not exist: %s", identifier);
+      } else {
+        return new BaseView(newViewOps(identifier), ViewUtil.fullViewName(name(), identifier));
+      }
+    }
+
+    throw new NoSuchViewException("Invalid view identifier: %s", identifier);
+  }
+
+  @Override
+  public ViewBuilder buildView(TableIdentifier identifier) {
+    return new BaseViewBuilder(identifier);
+  }
+
+  protected class BaseViewBuilder implements ViewBuilder {
+    private final TableIdentifier identifier;
+    private final ImmutableViewVersion.Builder viewVersionBuilder = ImmutableViewVersion.builder();
+    private final Map<String, String> properties = Maps.newHashMap();
+    private Schema schema;
+
+    protected BaseViewBuilder(TableIdentifier identifier) {
+      Preconditions.checkArgument(
+          isValidIdentifier(identifier), "Invalid view identifier: %s", identifier);
+      this.identifier = identifier;
+    }
+
+    @Override
+    public ViewBuilder withSchema(Schema newSchema) {
+      this.schema = newSchema;
+      viewVersionBuilder.schemaId(newSchema.schemaId());
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withQuery(String dialect, String sql) {
+      viewVersionBuilder.addRepresentations(
+          ImmutableSQLViewRepresentation.builder().dialect(dialect).sql(sql).build());
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withDefaultCatalog(String defaultCatalog) {
+      viewVersionBuilder.defaultCatalog(defaultCatalog);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withDefaultNamespace(Namespace namespace) {
+      viewVersionBuilder.defaultNamespace(namespace);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withProperties(Map<String, String> newProperties) {
+      this.properties.putAll(newProperties);
+      return this;
+    }
+
+    @Override
+    public ViewBuilder withProperty(String key, String value) {
+      this.properties.put(key, value);
+      return this;
+    }
+
+    @Override
+    public View create() {
+      ViewOperations ops = newViewOps(identifier);
+      if (null != ops.current()) {
+        throw new AlreadyExistsException("View already exists: %s", identifier);
+      }
+
+      ViewVersion viewVersion =
+          viewVersionBuilder
+              .versionId(1)
+              .timestampMillis(System.currentTimeMillis())
+              .putSummary("operation", "create")
+              .build();
+
+      ViewMetadata viewMetadata =
+          ViewMetadata.builder()
+              .setProperties(properties)
+              .setLocation(defaultWarehouseLocation(identifier))
```

Comment:
Why isn't it possible to set the location?

I think I asked about this elsewhere, but I don't see the reply. Sorry if it's a duplicate.

Reply:
sorry in case my answer got hidden by GH, but I did reply in https://github.com/apache/iceberg/pull/7880#discussion_r1325502327.

Reply:
Yes, it seems like an oversight that you can't set the location where a view is stored. I'd expect the builder to allow this, although it can be done in a follow up.

Reply:
I'll address this in a separate PR

Reply:
I've handled this in https://github.com/apache/iceberg/pull/8648

Reply:
Since this is an unresolved issue, I'm reopening the thread so that people can follow the link to the follow up PR.